fff

```diff
@@ -368,6 +369,49 @@ trash() {
     fi
 }
 
+bulk_rename() {
+  local tmp_file="$(mktemp)"
+
+  # Write all input file names to temporary file and open it in an editor.
+  printf "%s\n" "${@:1:$#-1}" > "$tmp_file"
+  "${EDITOR:-vi}" "$tmp_file"
+
+  local commands_file="$(mktemp)"
+
+  # Add instructions to the file with commands.
+  echo '# This file will be executed when you close the editor.' > "$commands_file"
+  echo '# Please double-check everything, clear the file to abort.' >> "$commands_file"
+
+  local counter=0
+  local changed=0
+  while read -r line || [ -n "$line" ] ; do
+    let counter+=1
+
+    # Trim read line.
+    line="${line#"${line%%[![:space:]]*}"}"
+    line="${line%"${line##*[![:space:]]}"}"
+
+    [ -z $line ] && continue
+
+    # Escape source and target arguments of mv command.
+    local source=$(printf "%q" "${@:$counter:1}")
```

Comment:
Can use `printf -v` instead.

`printf -v source '%q' "${@:counter:1}"`

Reply:
I don't think `local` is needed here as the variable is guaranteed to be overwritten each time the function is called. To avoid name conflicts the variables could be renamed to something relevant to the function name.